docs/api.adoc

```diff
@@ -341,6 +341,136 @@ Signs transaction offline with private key from keystore file with given passphr
 <-- {"jsonrpc": "2.0", "result": "0xd46e8dd67c5d32be8d46e8dd67c5d32be8058bb8eb970870f072445675058bb8eb970870f072445675", "id": 1}
 ----
 
+=== emerald_addContract
+
+Add a new contract and store the Application Binary Interface (ABI) locally.
+
+*Parameters*:
+
+    * `abi` (Array) - JSON format for a contract ABI, should be an array of function and/or event descriptions as defined https://github.com/ethereum/wiki/wiki/Ethereum-Contract-ABI[here]
+    ** `name` (String) - the name of the function
+    ** `inputs` (Array) - an array of objects, each of which contains a name and a type
+    ** `outputs` (Array) - an array of objects, each of which contains a name and a type
+    * `name` (String, optional)
+    * `address` (String) - hex-encoded 20 bytes public address
```

Comment:
if address is optional, then contracts should store the bytecode also. A contract ABI by itself is not useful.